src/main/java/se/sundsvall/precheck/service/PreCheckService.java

```diff
@@ -0,0 +1,59 @@
+package se.sundsvall.precheck.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.zalando.problem.Problem;
+import se.sundsvall.precheck.api.model.PreCheckResponse;
+import se.sundsvall.precheck.api.model.Status;
+import se.sundsvall.precheck.integration.citizen.CitizenIntegration;
+import se.sundsvall.precheck.integration.partyassets.PartyAssetsIntegration;
+
+import java.util.List;
+
+import static org.zalando.problem.Status.BAD_REQUEST;
+import static org.zalando.problem.Status.NOT_FOUND;
+import static se.sundsvall.precheck.constant.Constants.INVALID_MUNICIPALITY_ID;
+import static se.sundsvall.precheck.constant.Constants.NOT_FOUND_ERROR_MESSAGE;
+import static se.sundsvall.precheck.service.utils.PreCheckUtil.checkResourceAvailability;
+import static se.sundsvall.precheck.service.utils.PreCheckUtil.containsValidMunicipalityId;
+import static se.sundsvall.precheck.service.utils.PreCheckUtil.generateAssetTypeResponses;
+import static se.sundsvall.precheck.service.utils.PreCheckUtil.generateNoAssetTypeResponses;
+
+@Service
+public final class PreCheckService {
+
+    static final Logger LOGGER = LoggerFactory.getLogger(PreCheckService.class);
+    private final CitizenIntegration citizenIntegration;
+    private final PartyAssetsIntegration partyAssetsIntegration;
+
+
+    public PreCheckService(CitizenIntegration citizenIntegration, PartyAssetsIntegration partyAssetsIntegration) {
+        this.citizenIntegration = citizenIntegration;
+        this.partyAssetsIntegration = partyAssetsIntegration;
+    }
+
+    public ResponseEntity<List<PreCheckResponse>> checkPermit(String partyId, String municipalityId, final String assetType) {
```

Comment:
You increase the coupling in your code by doing this.
The service layer does not need to return a ResponseEntity. This type should only exist in the api-layer.

```
Some food for thought: 
What if we add a JMS interface to this service (or some other "non-HTTP" interface). Should that code be forced to deal with `ResponseEntities `when calling the service layer?
```